### PR TITLE
Fix price update

### PR DESCRIPTION
**Pull request by OlgaKochneva**
Fix problems with update-method in jpa repository.

### CODE REVIEW

File: src/main/java/com/epam/lowcost/controller/FlightController.java [hunk 1]
```diff
@@ -55,7 +58,12 @@ public String findFlightById(@PathVariable Long id, Model model) {
 
 
     @PostMapping(value = UPDATE)
-    public String updateFlight(@ModelAttribute("flight") Flight flight) {
+    public String updateFlight(@ModelAttribute("flight") Flight flight, BindingResult bindingResult, Model model) {
+        flightValidator.validate(flight, bindingResult);
+        if (bindingResult.hasErrors()) {
+            model.addAttribute("airports", airportService.getAllAirports());
+            return FLIGHTSETTINGS;
+        }
```
Comment (EugeneGrim) [inline src/main/java/com/epam/lowcost/controller/FlightController.java]: This method dosn't work: you always have at least one error in BindingResult

**Comment by EugeneGrim** [comment]
image
Flight search dosn't work when trying to buy a ticket

File: src/main/java/com/epam/lowcost/model/Ticket.java [hunk 2]
```diff
@@ -1,23 +1,20 @@
...
-import org.hibernate.annotations.Cascade;
+import lombok.*;
 
 import javax.persistence.*;
 import java.time.LocalDateTime;
 
 @Entity
 @Table(name = "TICKETS")
 @Data
+@Getter
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/model/Ticket.java]: getter exists in data, doesnt it ?

File: src/main/java/com/epam/lowcost/services/implementations/FlightServiceImpl.java [hunk 3]
```diff
@@ -73,7 +77,7 @@ public Flight deleteFlight(Long id) {
     @Override
     public List<Flight> getFilteredFlightsWithUpdatedPrice(Airport departureAirport, Airport arrivalAirport, LocalDateTime departureDateFrom, LocalDateTime departureDateTo) {
         List<Flight> flights = getByFromToDate(departureAirport, arrivalAirport, departureDateFrom, departureDateTo);
-        flights.forEach(f -> updateFlightPrice(f));
+        flights.forEach(f -> f.setInitialPrice(getUpdatedFlightPrice(f)));
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/services/implementations/FlightServiceImpl.java]: maybe u can use peek(...) in one stream?... I don't remember. And one terminal forEach at the end

File: src/main/java/com/epam/lowcost/controller/FlightController.java (hunk 1 above)
Comment (AnastasiaBerlina) [reply src/main/java/com/epam/lowcost/controller/FlightController.java]: I fixed it

**Pull request closed without merging**